Lib version test

GelberBaron opened the pull request:
In a Special Folder, I create a LIB Version for using the algorithm (code) without the program. It's now ready to be released.

GelberBaron committed:
Create main.cpp

- main file for new lib Version - implementing the main algorithm

GelberBaron committed:
Create GAUSSIAN.h

- new visible Header file (other Header file is only for Background, invisible functions)

GelberBaron committed:
Create background.h

- all "not public" function (invisible in stable LIB)

GelberBaron committed:
Create background.cpp

- all "not public" functions (invisible in later stable LIB / LIB release)

GelberBaron committed:
Delete background.cpp

- Copy it into main.cpp (use now keyword "static" to hide functions, so I have to merge it)

GelberBaron committed:
Update main.cpp

- Merge with "Background.cpp"

GelberBaron committed:
Update background.h

- Use keyword "static" to hide functions that are only local helpers

ljrk0 commented:
Willst du, dass ich das merge oder nicht?

GelberBaron commented:
Ich wollt noch eine Funktionalität einbauen, die mir gestern Abend eingefallen ist. Heute Mittag sollte ich commited haben, dann kannst du mergen.

ljrk0 commented:
du kannst das doch auch machen^^

ljrk0 commented:
du kannst glaub sogar rebase machen

GelberBaron commented:
Mal eine Frage am Rande: Kennt Linux das Prinzip der LIBs? (Zur Erläuterung: Das sind compilierte, aber nicht gelinkte Dateien, die man, wenn man die Includedatei hat, hinzu linken kann.)

ljrk0 commented:
logisch (.so)

GelberBaron commented:
Ich hab einen PullRequest / Branch gemacht, weil ich nicht ohne dein Einverständnis agieren wollte.

ljrk0 commented:
> Archive libraries (.a) are statically linked i.e when you compile your program with -c option in gcc. So, if there's any change in library, you need to compile and build your code again.
> 
> The advantage of .so (shared object) over .a library is that they are linked during the runtime i.e. after creation of your .o file -o option in gcc. So, if there's any change in .so file, you don't need to recompile your main program. But make sure that your main program is linked to the new .so file with > ln command.
> 
> This will help you to build the .so files. http://www.yolinux.com/TUTORIALS/LibraryArchives-StaticAndDynamic.html
> 
> Hope this helps.

from: http://stackoverflow.com/questions/9809213/what-are-a-and-so-files

GelberBaron commented:
Wenn Linux sowas hat, kannst du dann einen entsprechenen Build machen (wenn ich fertig bin)?

ljrk0 commented:
Habs noch nie gemacht. Werd sehen. brauchst du id. Libbranch main.cpp auch?

GelberBaron commented:
Nein, die main.cpp ist eine eigene, sie ist die main.cpp des LIB/A-Projects. Ich kenn das unter Windows so: 
- Implementierung von Funktionen in einem "LIB"-Projekt.
- Compilierung: Ein Compiler erzeugt eine LIB-Datei
- Alle _nicht als static deklarierten_ Funktionen sind in der LIB-Datei Verzeichnis
- In einem Projekt kann in den Linkereinstellungen die LIB hinzugefügt werden.
- Über eine .h-Datei (die normalerweise mit der LIB ausgeliefert wird) können alle _nicht als static deklarierten_ Funktionen im Quelltext benutzt werden.

Diese Vorgehensweise unterstützt sowohl MS Visual C++ als auch Pelles C Compiler, den ich für die Releases benutze.

GelberBaron commented:
Sorry, verklickt.

GelberBaron committed:
Update main.cpp

- add static keyword to in background.h as static declared functions
- now S-Matrix can be NULL

Don't Need the new "Background.h" (Change there only comments / documentation)

GelberBaron committed:
Update GAUSSIAN.h

- Add new feature description to documentation

GelberBaron committed:
Update main.cpp

- Bugfix

GelberBaron commented:
So, jetzt ist fertig. Wenn du einverstanden bist, merge ich.

ljrk0 commented:
logo
warum hast du das innnen eigne folder gemacht?
du hast doch ne egine branch O.o
